sympy/solvers/ode/tests/test_ode.py

```diff
@@ -33,6 +33,112 @@
 # dsolve() to return different results on different machines)
 
 
+def _ode_solver_test(ode_examples,our_hint='default'):
+    for example in ode_examples:
+        eq = ode_examples[example]['eq']
+        sol = ode_examples[example]['sol']
+        assert our_hint in classify_ode(eq), our_hint+"Hint was not recognised for the equation "+str(eq)
+        sols = sol + [constant_renumber(root) for root in sol]
+        sols += [root.expand() for root in sol]
+        expected_checkodesol = [(True, 0) for i in range(len(sol))]
+        assert dsolve(eq,hint=our_hint).rhs in sols, "Expected solution is different from solution returned by dsolve for the equation "+str(eq)
+        assert checkodesol(eq, sol) == expected_checkodesol, "The solution returned by dsolve is not correct for the equation "+str(eq)
```

Comment:
These lines are too long. The name of the equation e.g. `euler_hom_01` should be in the output of the error message.

Something like
```
dsolve_sol = dsolve(...)
if dsolve_sol != sol:
    message = """
Different solution found from dsolve for example {example}

The ODE is:
{eq}

The expected solution was
{sol}

What dsolve returned is
{dsolve_sol}
""".format(example=example, sol=sol, dsolve_sol=dsolve_sol)
    raise AssertionError(message)
```

Reply:
so should I do this for all three assert statements?

Reply:
Yes